server/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -551,12 +586,14 @@ private static NodeMetadata loadNodeMetadata(Settings settings, Logger logger, N
             final NodeMetadata legacyMetadata = NodeMetadata.FORMAT.loadLatestState(logger, NamedXContentRegistry.EMPTY, paths);
             if (legacyMetadata == null) {
                 assert nodeIds.isEmpty() : nodeIds;
-                metadata = new NodeMetadata(generateNodeId(settings), Version.CURRENT);
+                // If we couldn't find legacy metadata, we set the latest index version to this version
```

Comment:
Maybe mention that in this case we're starting a new empty node so no indices to worry about.